schematic/store/synapse.py

```diff
@@ -51,9 +52,101 @@
 from schematic.exceptions import MissingConfigValueError, AccessCredentialsError
 
 from schematic import CONFIG
+from schematic.utils.general import entity_type_mapping
 
 logger = logging.getLogger("Synapse storage")
 
+@dataclass
+class ManifestDownload(object):
+    """
+    syn: an object of type synapseclient.
+    manifest_id: id of a manifest  
+    """
+    syn: synapseclient.Synapse
+    manifest_id: str
+
+    def _download_manifest_to_folder(self):
+        """
+        try downloading a manifest to local cache or a given folder
+        manifest
+        Return: 
+            manifest_data: A new Synapse Entity object of the appropriate type
+        """
+         # TO DO: potentially deprecate the if else statement because "manifest_folder" key always exist in config
+        if CONFIG["synapse"]["manifest_folder"]:
+            manifest_data = self.syn.get(
+                    self.manifest_id,
+                    downloadLocation=CONFIG["synapse"]["manifest_folder"],
+                    ifcollision="overwrite.local",
+                )
+        else:
+            manifest_data = self.syn.get(
+                        self.manifest_id,
+                    )
+        return manifest_data 
+
+    def _entity_type_checking(self):
+        """
+        check the entity type of the id that needs to be downloaded
+        Return: 
+             if the entity type is wrong, raise an error
+        """
+        # check the type of entity
+        entity_type = entity_type_mapping(self.syn, self.manifest_id)
+        if entity_type  != "file":
+            logger.error(f'You are using entity type: {entity_type}. Please try using a file')
```

Comment:
@linglp This is minor but can you be more explicit in this error. Instead of using 'try', be more declarative, `Please use a file.' or 'Please provide a file id.'

Reply:
@mialy-defelice sounds good. I could make that change.